drivers/tee/optee/core.c

```diff
@@ -475,8 +448,44 @@ optee_config_shm_memremap(optee_invoke_fn *invoke_fn, void **memremaped_shm,
 		if (IS_ERR(rc))
 			goto err_memunmap;
 		priv_mgr = rc;
+
+		rc = optee_shm_pool_alloc_pages();
+		if (IS_ERR(rc))
+			goto err_free_priv_mgr;
+		dmabuf_mgr = rc;
 	} else {
-		const size_t sz = OPTEE_SHM_NUM_PRIV_PAGES * PAGE_SIZE;
+		/* Make inquiry about static SHM */
```

Comment:
It seems quite natural to move what's in this else block into a separate function.